tests/integration/security/testdata/authz/v1beta1-workload-ns1.yaml.tmpl

```diff
@@ -1,35 +1,18 @@
-# The following policy selects workload b in namespace 1
+# The following policy selects workload in namespace 1
```

Comment:
I think this is actually changing the coverage of the test.

In the old test, we intend to apply 3 policies `policy-ns1-b`, `policy-ns1-c` and `policy-ns1-x`, this is to test we are distributing the policies to the right workload, for example, we will send requests to verify `policy-ns1-b` is only in `b` but not `c`,  `policy-ns1-c` is in `c` but not `b` and `policy-ns1-x` is not in `b` nor `c`.

In the new test, it seems all policies are either applied to the same workload or not, and we are not testing difference policies on different workload. 

This is the special for `TestWorkloadSelector` that I think it makes it hard to be workload agnostic.

Reply:
How about I add `policy-ns1-c` back but replace `c` with `dst1`, and replace `dst` with `dst0`?

Reply:
@yangminzhu I have revised the workload selector test, please take a look.

Reply:
To cover the vm properly in the workload selector test, I think we should do the following:

Let's step back to look at the current policies:
- policy in root namespace: policy-ns-root-c
- policy in namespace 1: policy-ns1-b, policy-ns1-c, policy-ns1-x and policy-ns1-all
- policy in namespace 2: policy-ns2-c and policy-ns2-all

The current test has 3 workloads: bInNS1, cInNS1 and cInNS2

The current test cases are to check:
- bInNS1 only has policy-ns1-b and policy-ns1-all but not any other policies
- cInNS1 only has policy-ns1-all and policy-ns-root-c but not any other policies
- cInNS2 only has policy-ns2-c, policy-ns2-all and policy-ns-root-c but not any other policies

I'm actually not super clear what does it mean to support vm in this case? One thing is likely to replace (bInNS1, cInNS1 and cInNS2) with corresponding VMs (VMbInNS1, VMcInNS1 and VMcInNS2) and then repeat the test as a new set of test cases.

And this does not cover the case where we have both GKE and VM workloads, for example, policies only apply to GKE workloads but not VM workloads (or vice versa), this is a more complicated case and you likely need to write something new according to your vm test coverage requirements.

Reply:
The limitation is that there is one vm in ns1 in one cluster. We cannot replace bInNS1, cInNS1, and cInNS2 with VMs like vmAInNS1, vmBInNS1, and vmInNS2. I intend to apply same policies to pod b and vm, and keep the test coverage.

Reply:
the workload selector test is supposed to test multiple workloads with different labels in different namespace. If vm only has one workload in one namespace, it means it can not really be tested by this test. 

The problem is when you replace one of the k8s workloads to vm, you are reducing the existing test coverage (previously we runs all those cases in k8s and now only part of them are running for k8s). It is essentially having partial coverage of vm as well. Probably you should skip this one and create an issue for the test framework to support multiple VMs in different namespaces?

Reply:
By checking the policies in https://github.com/istio/istio/blob/master/tests/integration/security/testdata/authz/v1beta1-workload-ns1.yaml.tmpl, https://github.com/istio/istio/blob/master/tests/integration/security/testdata/authz/v1beta1-workload-ns2.yaml.tmpl and https://github.com/istio/istio/blob/master/tests/integration/security/testdata/authz/v1beta1-workload-ns-root.yaml.tmpl. The policies select b in ns1, c in ns1 and ns2. How about I add a similar policy for vm in ns1 like the existing b in ns1, that verifies the workload selector selects VM? In this way, we keep the coverage for k8s the same and add more for the VM. I also add a TODO into second test case about supporting multiple VMs in different namespaces.

Reply:
yes, you can add some new tests for vm without changing the current ones (except only the template part). We can always add more tests later but should not reduce current test coverage.

Reply:
Thanks, I have reverted the test cases back, and added a few cases for vm. Once we have more VMs available, I will refactor the test so that we replace all destination workloads with VMs. Please take a look.